components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -786,6 +810,26 @@ class GeckoEngineSession(
         }
     }
 
+    @VisibleForTesting
+    // Some site add extra information after the mimetype, for example 'application/pdf; qs=0.001'
+    // we just want to extract the mimeType and ignore the rest
+    internal fun extractMimeType(contentType: String?): String? {
+        return if (contentType != null) {
+            if (contentType.contains(";")) {
+                contentType.substringBefore(";")
+            } else {
+                contentType
+            }
+        } else {
+            null
+        }
+    }
```

Comment:
I wonder if this is something `guessFileName()` should do whenever we give a mimetype header.

Reply:
I guess this is a problem that we are encountering because of reading raw data from headers and in  `guessFileName` is expecting clean `contentType` data. What do you think?

Reply:
Hm, okay. It also takes `contentDisposition` which is a header, maybe the other parameter should also take be able to handle the raw header? 🤔

Reply:
I updated the pr and added as part of `guessFileName()`, we will need to do a follow up for it https://github.com/mozilla-mobile/android-components/issues/8425